standardized-output-composition.md

```diff
@@ -0,0 +1,291 @@
+# Output formatting
+
+## Summary
+
+JSON Schema is defined to be platform-independent.  As such, to increase compatibility across platforms, implementations SHOULD conform to a standard validation output format.  This section describes the minimum requirements that consumers will need to properly interpret validation results.
+
+## Format
+
+JSON Schema output is defined using JSON.  Implementations MAY deviate from this as supported by their specific languages and platforms, however it is RECOMMENDED that the output be convertible to the JSON format defined herein via serialization or other means.
+
+## Output Levels **<span style="color:red">(is "levels" the right word?)</span>**
+
+This specification defines four levels of output.  See the *Output Structure* section for the requirements of each level.
+
+- **Pass/Fail** - A boolean which simply indicates whether the schema was validated.
+- **Basic** - Provides validation information in a flat list structure.
+- **Detailed** - Provides validation information in a condensed hierarchical structure based on the structure of the schema.
+- **Verbose** - Provides validation information in an uncondensed hierarchical structure that matches the exact structure of the schema.
+
+Implementations SHOULD specify which levels of validation they support.
+
+## Minimum Information
+
+Beyond a simplistic pass/fail, additional information is useful to aid in debugging a schema or instance.  Each sub-result SHOULD contain the information contained within this section at a minimum.
+
+A single object that contains all of these components is considered an output unit.
+
+Implementations MAY elect to provide additional information.
+
+### Keyword relative location
+
+The relative location of the validating keyword that follows the validation path.  The value MUST be expressed as a JSON Pointer, and it MUST include `$ref` or `$recursiveRef` segments for any reference jumps **<span style="color:red">(wording?)</span>**.
```

Comment:
Including this was one of @handrews's ideas (and a good one). He felt it necessary to not only include the route to the subschema that is resolved but also the route that validation took to get there.  That there may be multiple paths that resolve to the same location is not really a concern.

I suppose in that regard it would strictly be a JSON Pointer, but it's understandable within the context of processing a schema.  Additionally, when a referencing segment is included, a direct URI to the subschema is also required so that the consumer of the output doesn't have to do the processing you mentioned.

Reply:
I'm making a new comment based on this.